### PR TITLE
Refactor/get persons by nickname pagination

### LINKED ISSUE #277: Add get_persons_by_nickname with pagination to TahrirDatabase
Add get_persons_by_nickname method to TahrirDatabase to support paginated user search by nickname.

This moves the user search and pagination logic from the tahrir frontend into the API layer where it belongs. The method accepts a search string, begin offset, and limit, and returns a dict with the matching users, total count, and pagination info.

### PR DESCRIPTION
**Pull request by Daniel-1600**
solves #277

I added the logic query for the database to the api where the database does the slicing itself rather than python on the endpoint. this is more efficient since you will not have to fetch all the results after you search something it just narrows down to a number until you get what you want.

here is a short example of it in play 
image

### CODE REVIEW

File: tahrir_api/dbapi.py [hunk 1]
```diff
@@ -531,6 +531,43 @@ def get_all_persons(self, include_opted_out=False):
             query = query.filter(not_(Person.opt_out))
         return query
 
+    def get_persons_by_nickname(self, search_string: str, begin: int = 0, limit: int = 100):
+        """
+        Search for persons by nickname with pagination.
+        :type search_string: str
```
Comment (sdglitched) [inline tahrir_api/dbapi.py:538]:
```suggestion
        
        :type search_string: str
```

File: tahrir_api/dbapi.py [hunk 2]
```diff
@@ -531,6 +531,43 @@ def get_all_persons(self, include_opted_out=False):
...
+        :param search_string: The nickname pattern to search for.
+        :type begin: int
+        :param begin: Offset for pagination (default 0).
+        :type limit: int
+        :param limit: Max results per page, capped at 100 (default 100).
+        """
+        safe_limit = min(limit, 100)
+        query = self.session.query(Person).filter(
```
Comment (sdglitched) [inline tahrir_api/dbapi.py:548]:
```suggestion
        
        query = self.session.query(Person).filter(
```

File: tahrir_api/dbapi.py [hunk 3]
```diff
@@ -531,6 +531,43 @@ def get_all_persons(self, include_opted_out=False):
...
+            func.lower(Person.nickname).like(f"%{search_string.lower()}%")
+        )
+        total_count = query.count()
```
Comment (sdglitched) [inline tahrir_api/dbapi.py:552]:
```suggestion

        total_count = query.count()
```

File: tahrir_api/dbapi.py [hunk 4]
```diff
@@ -531,6 +531,43 @@ def get_all_persons(self, include_opted_out=False):
...
+                "created_on": person.created_on.timestamp() if person.created_on else None,
+                "email": person.avatar,  # the endpoint handles the hashing
+                "last_login": person.last_login.timestamp() if person.last_login else None,
+                "nickname": person.nickname,
+                "opt_out": person.opt_out,
+                "rank": person.rank,
+                "website": person.website,
+            }
+            for person in paginated_results
+        ]
```
Comment (sdglitched) [inline tahrir_api/dbapi.py]: As mentioned previously, this should not be applied in the API layer. Please remove it.

File: tests/test_dbapi.py [hunk 5]
```diff
@@ -675,3 +675,43 @@ def test_get_all_series_empty(api):
...
+    api.add_person("test@test.com", nickname="test_user")
+    result = api.get_persons_by_nickname("zzznomatch")
+    assert result["total"] == 0
+    assert result["users"] == []
+
+
+def test_get_persons_by_nickname_case_insensitive(api):
+    api.add_person("upper@test.com", nickname="UpperCase")
+    result = api.get_persons_by_nickname("uppercase")
+    assert result["total"] == 1
```
Comment (sdglitched) [inline tests/test_dbapi.py]: As mentioned earlier, I would want you to parameterise the tests into two parts. One for the general search, which will include all the edge cases for the searching string, and another one for the pagination.

**Review by sdglitched — CHANGES_REQUESTED** [review]
Please make requested changes.

**Comment by Daniel-1600** [comment]
> Please make requested changes.

I’ve addressed all the requested changes. Please let me know if everything looks good.

**Review by sdglitched — APPROVED** [review]
LGTM 🚀 

@Daniel-1600, thank you for working on this.

@gridhead, we can merge this PR.

**Comment by sdglitched** [comment]
@gridhead, this PR is ready for merging. Once you are around and available, please take a look.

**Review by gridhead — APPROVED** [review]
LGTM

Thanks for the review, @sdglitched.

Thanks for the changes, @Daniel-1600.
